lib/datadog/di/code_tracker.rb

```diff
@@ -0,0 +1,152 @@
+# frozen_string_literal: true
+
+module Datadog
+  module DI
+    # Tracks loaded Ruby code by source file and maintains a map from
+    # source file to the loaded code (instruction sequences).
+    # Also arranges for code in the loaded files to be instrumented by
+    # line probes that have already been received by the library.
+    #
+    # The loaded code is used to target line trace points when installing
+    # line probes which dramatically improves efficiency of line trace points.
+    #
+    # Note that, since most files will only be loaded one time (via the
+    # "require" mechanism), the code tracker needs to be global and not be
+    # recreated when the DI component is created.
+    #
+    # @api private
+    class CodeTracker
+      def initialize
+        @registry = Hash.new
+        @trace_point_lock = Mutex.new
+        @registry_lock = Hash.new
+      end
+
```

Comment:
I suggest adding, even if a very simple, method level documentation to the `start` method.

Reply:
Yes, i will type up a comment.